### DISCUSSION TITLE
Can stubs be used to hint PHPStorm?

### DISCUSSION THREAD

**Discussion by jameshulse** [larastan/larastan#1815] [General]
Hi, I'm new to the Larastan project but I came across the 'stubs' which look really complete. Is it possible to use these in PHPStorm to improve Laravel type hinting? Is this their purpose? Or are they used by Larastan for it's analysis?

Thanks!

**Comment by calebdw** [comment]
Take a look at the [laravel-ide-helper](https://github.com/barryvdh/laravel-ide-helper) package. The stubs are used to override the vendor type hints for the analysis.

**Reply by jameshulse** [reply]
Thanks. We use laravel-ide-helper, but there are still some issues with Laravel query builder functions not knowing that they return an instance of the model. Larastan's stubs look to address that by introducing templating to the builder doc strings.

**Comment by szepeviktor** [comment]
You may want to read this https://blog.jetbrains.com/phpstorm/2018/03/how-to-provide-stubs-for-phpstorm/

**Reply by jameshulse** [reply]
Thanks. I did find this so might try it myself - but I was wondering how other people were managing this, or if they have created a public plugin themselves?

**Reply by jameshulse** [reply]
Update: I tried this, and can't get any improvements to my intellisense. I'll keep digging - maybe an update to Laravel needs to be made directly to support templates on it's classes.
